core/src/main/java/org/kiwix/kiwixmobile/core/main/NightModeViewPainter.kt

```diff
@@ -0,0 +1,96 @@
+/*
+ * Kiwix Android
+ * Copyright (c) 2020 Kiwix <android.kiwix.org>
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+package org.kiwix.kiwixmobile.core.main
+
+import android.content.Context
+import android.graphics.ColorMatrixColorFilter
+import android.graphics.Paint
+import android.view.View
+import android.view.ViewGroup
+import android.webkit.WebView
+import org.kiwix.kiwixmobile.core.NightModeConfig
+import org.kiwix.kiwixmobile.core.utils.SharedPreferenceUtil
+import javax.inject.Inject
+
+/**
+ * NightModeViewPainter class is used to apply respective filters to the views
+ * depending whether the app is in dark mode or not
+ * Created by yashk2000 on 24/03/2020.
+ */
+
+class NightModeViewPainter @Inject constructor(
+  val context: Context?
+) {
+
+  private var nightModeConfig: NightModeConfig? = null
+  private val invertedPaint = createInvertedPaint()
+
+  fun update(view: View) {
+    val sharedPreferenceUtil = SharedPreferenceUtil(context)
```

Comment:
sharedPref and NightModeConfig should be in the constructor

Reply:
@macgills I if I shift NightModeConfig in the constructor, we won't need context or sharedPref. We only needed them for getting nightModeConfig. So I'll do that.